checker/jdk/nullness/src/java/util/EnumMap.java

```diff
@@ -0,0 +1,839 @@
+     * then this method returns {@code v}; otherwise it returns
+     * {@code null}.  (There can be at most one such mapping.)
+     *
+     * <p>A return value of {@code null} does not <i>necessarily</i>
+     * indicate that the map contains no mapping for the key; it's also
+     * possible that the map explicitly maps the key to {@code null}.
+     * The {@link #containsKey containsKey} operation may be used to
+     * distinguish these two cases.
+     */
+    public @Nullable V get(@Nullable Object key) {
+        return (isValidKey(key) ?
+                unmaskNull(vals[((Enum<?>)key).ordinal()]) : null);
+    }
+
+    // Modification Operations
+
+    /**
+     * Associates the specified value with the specified key in this map.
+     * If the map previously contained a mapping for this key, the old
+     * value is replaced.
+     *
+     * @param key the key with which the specified value is to be associated
+     * @param value the value to be associated with the specified key
+     *
+     * @return the previous value associated with specified key, or
+     *     <tt>null</tt> if there was no mapping for key.  (A <tt>null</tt>
+     *     return can also indicate that the map previously associated
+     *     <tt>null</tt> with the specified key.)
+     * @throws NullPointerException if the specified key is null
+     */
+    public @Nullable V put(K key, V value) {
+        typeCheck(key);
+
+        int index = key.ordinal();
+        Object oldValue = vals[index];
+        vals[index] = maskNull(value);
+        if (oldValue == null)
+            size++;
+        return unmaskNull(oldValue);
+    }
+
+    /**
+     * Removes the mapping for this key from this map if present.
+     *
+     * @param key the key whose mapping is to be removed from the map
+     * @return the previous value associated with specified key, or
+     *     <tt>null</tt> if there was no entry for key.  (A <tt>null</tt>
+     *     return can also indicate that the map previously associated
+     *     <tt>null</tt> with the specified key.)
+     */
+    public @Nullable V remove(@Nullable Object key) {
+        if (!isValidKey(key))
+            return null;
+        int index = ((Enum<?>)key).ordinal();
+        Object oldValue = vals[index];
+        vals[index] = null;
+        if (oldValue != null)
+            size--;
+        return unmaskNull(oldValue);
+    }
+
+    private boolean removeMapping(@Nullable Object key, @Nullable Object value) {
+        if (!isValidKey(key))
+            return false;
+        int index = ((Enum<?>)key).ordinal();
+        if (maskNull(value).equals(vals[index])) {
+            vals[index] = null;
+            size--;
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Returns true if key is of the proper type to be a key in this
+     * enum map.
+     */
+    @EnsuresNonNullIf(expression="#1", result=true)
+    private boolean isValidKey(Object key) {
```

Comment:
Should `key` also be annotated as `@Nullable`?  Otherwise a call with a possibly-null value is not permitted, and the `@EnsuresNonNullIf` annotation is moot.